**Linux** [Build 9448](http://brontes.lab.eng.brq.redhat.com/viewLog.html?buildId=91347&buildTypeId=WFPR) outcome was **FAILURE** using a merge of e916c3e7a71bfbba523df5703acafd8a9ef2dd1e
Summary: Tests failed: 1 (1 new), passed: 3306, ignored: 193 Build time: 01:13:15
### Failed tests
```
org.jboss.as.test.integration.ejb.mdb.ejb2x.MDB20TopicTestCase.testEjb20TopicMDBs: java.lang.AssertionError: Reply message was null on reply queue: ActiveMQQueue[ejb2x/replyQueueA]
	at org.jboss.as.test.integration.ejb.mdb.ejb2x.MDB20TopicTestCase.testEjb20TopicMDBs(MDB20TopicTestCase.java:121)
------- Stdout: -------
 [0m01:55:11,566 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-1) WFLYSRV0028: Stopped deployment mdb.jar (runtime-name: mdb.jar) in 36ms
 [0m [0m01:55:11,571 INFO  [org.jboss.as.repository] (management-handler-thread - 4) WFLYDR0002: Content removed from location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/basic/target/jbossas/standalone/data/content/ef/3772d960efb9e052fc766cd209cf9d89206353/content
 [0m [0m01:55:11,571 INFO  [org.jboss.as.server] (management-handler-thread - 4) WFLYSRV0009: Undeployed "mdb.jar" (runtime-name: "mdb.jar")
 [0m01:55:11,594 INFO  [org.jboss.as.test.integration.common.jms.JMSOperationsProvider] (main) Operation successful for update = {
    "operation" => "remove",
    "address" => [
        ("subsystem" => "messaging-activemq"),
        ("server" => "default"),
        ("jms-queue" => "ejb2x/queue")
    ]
}
01:55:11,609 INFO  [org.jboss.as.test.integration.common.jms.JMSOperationsProvider] (main) Operation successful for update = {
    "operation" => "remove",
    "address" => [
        ("subsystem" => "messaging-activemq"),
        ("server" => "default"),
        ("jms-queue" => "ejb2x/replyQueue")
    ]
}
01:55:11,618 INFO  [org.jboss.as.test.integration.ejb.mdb.ejb2x.AbstractMDB2xTestCase] (main) MDB20TopicTestCase.jar:
/org/
/org/jboss/
/org/jboss/as/
/org/jboss/as/test/
/org/jboss/as/test/integration/
/org/jboss/as/test/integration/ejb/
/org/jboss/as/test/integration/ejb/mdb/
/org/jboss/as/test/integration/ejb/mdb/ejb2x/
/org/jboss/as/test/integration/ejb/mdb/ejb2x/MDB20TopicTestCase$JmsQueueSetup.class
/org/jboss/as/test/integration/ejb/mdb/ejb2x/EJB2xMDB.class
/org/jboss/as/test/integration/ejb/mdb/ejb2x/AbstractMDB2xTestCase.class
/org/jboss/as/test/integration/common/
/org/jboss/as/test/integration/common/jms/
/org/jboss/as/test/integration/common/jms/JMSOperations.class
/org/jboss/as/test/integration/common/jms/JMSOperationsProvider.class
/org/jboss/as/test/integration/common/jms/ActiveMQProviderJMSOperations.class
/org/jboss/as/test/integration/common/jms/JMSOperationsException.class
/org/jboss/as/test/shared/
/org/jboss/as/test/shared/TimeoutUtil.class
/META-INF/
/META-INF/MANIFEST.MF
/META-INF/jboss-permissions.xml
/META-INF/ejb-jar.xml
/META-INF/jboss-ejb3.xml
 [0m01:55:11,644 INFO  [org.apache.activemq.artemis.core.server] (ServerService Thread Pool -- 16) AMQ221003: trying to deploy queue jms.topic.ejb2x/topic
 [0m01:55:11,679 INFO  [org.jboss.as.test.integration.common.jms.JMSOperationsProvider] (main) Operation successful for update = {
    "operation" => "add",
    "address" => [


```

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/HTTPUpgradeService.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/HTTPUpgradeService.java
@@ -54,6 +54,7 @@ import org.jboss.msc.service.StopContext;
 import org.jboss.msc.value.InjectedValue;
 import org.wildfly.extension.messaging.activemq.logging.MessagingLogger;
 import org.xnio.ChannelListener;
+import org.xnio.IoUtils;
 import org.xnio.StreamConnection;
 import org.xnio.netty.transport.WrappingXnioSocketChannel;
 
@@ -147,10 +148,14 @@ public class HTTPUpgradeService implements Service<HTTPUpgradeService> {
             @Override
             public void handleEvent(final StreamConnection connection) {
                 MessagingLogger.ROOT_LOGGER.debugf("Switching to %s protocol for %s http-acceptor", protocolName, acceptorName);
-                SocketChannel channel = new WrappingXnioSocketChannel(connection);
                 RemotingService remotingService = activemqServer.getRemotingService();
-
-                NettyAcceptor acceptor = (NettyAcceptor)remotingService.getAcceptor(acceptorName);
+                if (!remotingService.isStarted()) {
+                    // ActiveMQ no longer accepts connection
+                    IoUtils.safeClose(connection);
+                    return;
+                }
+                NettyAcceptor acceptor = (NettyAcceptor) remotingService.getAcceptor(acceptorName);
+                SocketChannel channel = new WrappingXnioSocketChannel(connection);
                 acceptor.transfer(channel);
                 connection.getSourceChannel().resumeReads();
             }
@@ -178,8 +183,7 @@ public class HTTPUpgradeService implements Service<HTTPUpgradeService> {
      * Service to handle HTTP upgrade for legacy (HornetQ) clients.
      *
      * Legacy clients use different protocol and security key and accept headers during the HTTP Upgrade handshake.
-     */
-    static class LegacyHttpUpgradeService extends HTTPUpgradeService {
+     */static class LegacyHttpUpgradeService extends HTTPUpgradeService {
 
         public static void installService(final ServiceTarget serviceTarget, String activeMQServerName, final String acceptorName, final String httpListenerName) {
 

